lint/tests/test_lint.py

```diff
@@ -153,6 +153,72 @@ def test_lint_failing(capsys):
     assert err == ""
 
 
+def test_ref_existent_relative(capsys):
+    with _mock_lint("check_path") as mocked_check_path:
+        with _mock_lint("check_file_contents") as mocked_check_file_contents:
+            rv = lint(_dummy_repo, ["ref/existent_relative.html"], False, False)
+            assert rv == 0
+            assert mocked_check_path.call_count == 1
+            assert mocked_check_file_contents.call_count == 1
+    out, err = capsys.readouterr()
+    assert out == ""
+    assert err == ""
+
+
+def test_ref_existent_root_relative(capsys):
+    with _mock_lint("check_path") as mocked_check_path:
+        with _mock_lint("check_file_contents") as mocked_check_file_contents:
+            rv = lint(_dummy_repo, ["ref/existent_root_relative.html"], False, False)
+            assert rv == 0
+            assert mocked_check_path.call_count == 1
+            assert mocked_check_file_contents.call_count == 1
+    out, err = capsys.readouterr()
+    assert out == ""
+    assert err == ""
+
+
+def test_ref_non_existent_relative(capsys):
+    with _mock_lint("check_path") as mocked_check_path:
+        with _mock_lint("check_file_contents") as mocked_check_file_contents:
+            rv = lint(_dummy_repo, ["ref/non_existent_relative.html"], False, False)
+            assert rv == 1
+            assert mocked_check_path.call_count == 1
+            assert mocked_check_file_contents.call_count == 1
+    out, err = capsys.readouterr()
+    assert "NON-EXISTENT-REF" in out
+    assert "ref/non_existent_relative.html" in out
+    assert "non_existent_file.html" in out
+    assert err == ""
+
+
+def test_ref_non_existent_root_relative(capsys):
+    with _mock_lint("check_path") as mocked_check_path:
+        with _mock_lint("check_file_contents") as mocked_check_file_contents:
+            rv = lint(_dummy_repo, ["ref/non_existent_root_relative.html"], False, False)
+            assert rv == 1
+            assert mocked_check_path.call_count == 1
+            assert mocked_check_file_contents.call_count == 1
+    out, err = capsys.readouterr()
+    assert "NON-EXISTENT-REF" in out
+    assert "ref/non_existent_root_relative.html" in out
+    assert "/non_existent_file.html" in out
+    assert err == ""
+
+
+def test_ref_absolute_url(capsys):
```

Comment:
This was passing locally. It took me much too long to realize that I had forgotten to stag the requisite fixture file. Anyway, the tests are passing in CI now.